src/script/miniscript.h

```diff
@@ -1030,29 +1030,39 @@ inline NodeRef<Key> Parse(Span<const char> in, const Ctx& ctx)
 {
     using namespace spanparsing;
 
+    // Account for the number of fragments we started to parse. Any fragment is at least one byte
+    // long. If we started to parse more than 3600 of them, don't bother continuing: the resulting
+    // Miniscript would be invalid.
+    size_t opened_frags = 0;
+
     // The two integers are used to hold state for thresh()
     std::vector<std::tuple<ParseContext, int64_t, int64_t>> to_parse;
     std::vector<NodeRef<Key>> constructed;
 
     to_parse.emplace_back(ParseContext::WRAPPED_EXPR, -1, -1);
 
     while (!to_parse.empty()) {
+        if (opened_frags > MAX_STANDARD_P2WSH_SCRIPT_SIZE) return {};
+
         // Get the current context we are decoding within
         auto [cur_context, n, k] = to_parse.back();
         to_parse.pop_back();
 
         switch (cur_context) {
         case ParseContext::WRAPPED_EXPR: {
-            int colon_index = -1;
-            for (int i = 1; i < (int)in.size(); ++i) {
+            std::optional<size_t> colon_index{};
```

Comment:
In 088e5b0b9317bd1d7c9755fb839b7326d99bc910 "miniscript: upper bound on the size of inputs to consider for parsing"

It seems like this is an unrelated change. I don't see how changing `colon_index` to an optional is related to the size check.

Reply:
In order to avoid explicit cast all around the place i made it a `size_t`, which makes it impossible to use the `-1` magic value. Hence the optional.